redwood-lazylayout-uiview/src/commonMain/kotlin/app/cash/redwood/lazylayout/uiview/UIViewLazyList.kt

```diff
@@ -25,133 +25,144 @@ import app.cash.redwood.Modifier
+    }
 
-      override fun collectionView(
-        collectionView: UICollectionView,
-        cellForItemAtIndexPath: NSIndexPath,
-      ): UICollectionViewCell {
-        val cell = collectionView.dequeueReusableCellWithReuseIdentifier(
-          identifier = reuseIdentifier,
-          forIndexPath = cellForItemAtIndexPath,
-        ) as LazyListContainerCell
+    override fun deleteRows(index: Int, count: Int) {
+      // TODO(jwilson): pass a range somehow when 'count' is large?
+      tableView.deleteRowsAtIndexPaths(
+        (index until index + count).map { NSIndexPath.indexPathForItem(it.convert(), 0) },
+        UITableViewRowAnimationNone,
+      )
+    }
+  }
 
-        cell.collectionViewFrame = collectionView.frame
-        cell.collectionViewFlowLayout = collectionViewFlowLayout
+  override val placeholder: Widget.Children<UIView> = processor.placeholder
 
-        val widget = items.itemForGlobalIndex(cellForItemAtIndexPath.item.toInt())
-        cell.set(widget.value)
+  override val items: Widget.Children<UIView> = processor.items
 
-        return cell
-      }
-    }
+  private val dataSource = object : NSObject(), UITableViewDataSourceProtocol {
+    override fun tableView(
+      tableView: UITableView,
+      numberOfRowsInSection: NSInteger,
+    ) = processor.size.toLong()
 
-  private val collectionViewDelegate: UICollectionViewDelegateFlowLayoutProtocol =
-    object : NSObject(), UICollectionViewDelegateFlowLayoutProtocol {
-      override fun collectionView(
-        collectionView: UICollectionView,
-        layout: UICollectionViewLayout,
-        minimumLineSpacingForSectionAtIndex: NSInteger,
-      ): CGFloat {
-        return 0.0
-      }
+    override fun tableView(
+      tableView: UITableView,
+      cellForRowAtIndexPath: NSIndexPath,
+    ) = processor.getCell(cellForRowAtIndexPath.item.toInt())
+  }
 
+  private val tableViewDelegate: UITableViewDelegateProtocol =
+    object : NSObject(), UITableViewDelegateProtocol {
       override fun scrollViewDidScroll(scrollView: UIScrollView) {
-        val visibleIndexPaths = collectionView.indexPathsForVisibleItems()
+        val visibleIndexPaths = tableView.indexPathsForVisibleRows ?: return
 
         if (visibleIndexPaths.isNotEmpty()) {
           // TODO: Optimize this for less operations
-          updateViewport(
+          onViewportChanged?.invoke(
             visibleIndexPaths.minOf { (it as NSIndexPath).item.toInt() },
             visibleIndexPaths.maxOf { (it as NSIndexPath).item.toInt() },
           )
         }
       }
     }
 
-  override val placeholder = MutableListChildren<UIView>()
-
   init {
-    collectionView.apply {
-      dataSource = collectionViewDataSource
-      delegate = collectionViewDelegate
+    tableView.apply {
+      dataSource = this@UIViewLazyList.dataSource
+      delegate = tableViewDelegate
       prefetchingEnabled = true
+      rowHeight = UITableViewAutomaticDimension
+      estimatedRowHeight = 44.0
```

Comment:
nit: is this needed? 44 is the default value since iOS 11

Reply:
Fixed!